### PR TITLE
Added caching to GitHub connector.

### PR DESCRIPTION
**Pull request by diminishedprime**
Adding a cache for the GitHub.com community connector.

### CODE REVIEW

File: community-connectors/github/CustomCache.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  if (sheet === null) {
+    spreadsheet.insertSheet(sheetName);
+  }
+  this.sheet = spreadsheet.getSheetByName(sheetName);
+  return this;
+}
+
+CustomCache.SCRIPT_ID_KEY = 'sheets cache script id';
+
+CustomCache.SPREADSHEET_NAME = 'user cache for GitHub.com connector.';
```
Comment (minhaz) [inline community-connectors/github/CustomCache.js]: If the CustomCache.js is reusable, should this be a more generic name or defined in the main module?

File: community-connectors/github/CustomCache.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+ * Returns the range for the sheet.
+ *
+ * @return {object} the range for the sheet.
+ */
+CustomCache.prototype.getEntireRange = function() {
+  var lastRow = Math.max(this.sheet.getLastRow(), 1);
+  return this.sheet.getRange(1, 1, lastRow, 2);
+};
+
+CustomCache.prototype.findIndex = function(predicate) {
```
Comment (minhaz) [inline community-connectors/github/CustomCache.js]: JSDoc?

File: community-connectors/github/main.js [hunk 3]
```diff
@@ -316,11 +274,129 @@ connector.rowifyStarData = function(stars, dataSchema) {
...
+  if (cached) {
+    return cached;
+  } else {
+    try {
+      var shouldCache = true;
+      var response = connector.getUrlFetchApp().fetch(nextUrl, options);
+      var results = JSON.parse(response);
+      nextUrl = connector.getNextLink(response);
+
+      if (!nextUrl || results.length !== 100) {
```
Comment (minhaz) [inline community-connectors/github/main.js]: Should this be hardcoded?

File: community-connectors/github/CustomCache.js (hunk 2 above)
Comment (diminishedprime) [reply community-connectors/github/CustomCache.js]: done

File: community-connectors/github/CustomCache.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+CustomCache.getSpreadsheetAppNS = function() {
+  return SpreadsheetApp;
+};
+
+CustomCache.getPropertiesServiceNS = function() {
+  return PropertiesService;
+};
+
+function clearUserProperties() {
```
[the remark this answers is not in the document]
Comment (diminishedprime) [reply community-connectors/github/CustomCache.js]: This is not used anywhere, but it's the most convenient way to invalidate the cache. I agree that it should only remove properties that are used for the cache. Will fix that.

File: community-connectors/github/CustomCache.js (hunk 1 above)
Comment (diminishedprime) [reply community-connectors/github/CustomCache.js]: Right now, the CustomCache isn't quite ready to be reusable. Until I know which parts make sense to keep between different connectors, I'm planning to punt on things like this.

File: community-connectors/github/main.js [hunk 5]
```diff
@@ -193,82 +203,30 @@ connector.getSchema = function(request) {
  */
 connector.validateConfig = function(configParams) {
   configParams = configParams || {};
-
-  configParams.Organization = configParams.Organization ||
-      connector.throwError('You must provide an Organization.', true);
-
-  configParams.Repository = configParams.Repository ||
-      connector.throwError('You must provide a Repository.', true);
+  configParams.Organization = configParams.Organization || 'google';
```
[the remark this answers is not in the document]
Comment (diminishedprime) [reply community-connectors/github/main.js]: Done.

File: community-connectors/github/main.js (hunk 3 above)
Comment (diminishedprime) [reply community-connectors/github/main.js]: Done.

**Pull request merged**